packages/datadog-plugin-google-cloud-vertexai/src/index.js

```diff
@@ -0,0 +1,195 @@
+'use strict'
+
+const { MEASURED } = require('../../../ext/tags')
+const { storage } = require('../../datadog-core')
+const TracingPlugin = require('../../dd-trace/src/plugins/tracing')
+const makeUtilities = require('../../dd-trace/src/plugins/util/llm')
+
+class GoogleCloudVertexAIPlugin extends TracingPlugin {
+  static get id () { return 'google-cloud-vertexai' }
+  static get prefix () {
+    return 'tracing:apm:vertexai:request'
+  }
+
+  constructor () {
+    super(...arguments)
+
+    Object.assign(this, makeUtilities('vertexai', this._tracerConfig))
+  }
+
+  bindStart (ctx) {
+    const { instance, request, resource, stream } = ctx
+
+    const tags = this.tagRequest(request, instance, stream)
+
+    const span = this.startSpan('vertexai.request', {
+      service: this.config.service,
+      resource,
+      kind: 'client',
+      meta: {
+        [MEASURED]: 1,
+        ...tags
+      }
+    }, false)
+
+    const store = storage('legacy').getStore() || {}
+    ctx.currentStore = { ...store, span }
+
+    return ctx.currentStore
+  }
+
+  asyncEnd (ctx) {
+    const span = ctx.currentStore?.span
+    if (!span) return
+
+    const { result } = ctx
+
+    const response = result?.response
+    if (response) {
+      const tags = this.tagResponse(response)
+      span.addTags(tags)
+    }
+
+    span.finish()
+  }
+
+  tagRequest (request, instance, stream) {
+    const model = extractModel(instance)
+    const tags = {
+      'vertexai.request.model': model
+    }
+
+    const history = instance.historyInternal
+    let contents = typeof request === 'string' || Array.isArray(request) ? request : request.contents
+    if (history) {
+      contents = [...history, ...(Array.isArray(contents) ? contents : [contents])]
+    }
+
+    const generationConfig = instance.generationConfig || {}
+    for (const key of Object.keys(generationConfig)) {
+      const transformedKey = key.replace(/([a-z0-9])([A-Z])/g, '$1_$2').toLowerCase()
+      tags[`vertexai.request.generation_config.${transformedKey}`] = JSON.stringify(generationConfig[key])
```

Comment:
do we need to JSON.stringify each config setting value? Can't we just directly cast as a string instead?

Reply:
it wouldn't work for object-type configurations. For example, `responseSchema`. casting it to a string would yield `[object Object]`, while `JSON.stringify` would stringify it as `'{"type":...,"properties":...,"required":...'` (or however it may look):
```
> const obj = { a: 5 }
> String(obj)
'[object Object]'
> JSON.stringify(obj)
'{"a":5}'
```
i actually think besides `responseSchema` everything would be fine casting as a string. I don't think the `JSON.stringify` has too much of a perf impact over just casting to a String directly. let me know your thoughts, we could just special-case `responseSchema` off of the tags!

Reply:
Ahh thanks for explaining. This is fine for me as-is if you don't think the performance impact is that significant.